x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResponseListener.java

```diff
@@ -154,23 +154,31 @@ private RestResponse buildResponse(EsqlQueryResponse esqlResponse) throws IOExce
     }
 
     /**
-     * Log the execution time and query when handling an ES|QL response.
+     * Log internal server errors all the time and log queries if debug is enabled.
      */
     public ActionListener<EsqlQueryResponse> wrapWithLogging() {
+        ActionListener<EsqlQueryResponse> listener = ActionListener.wrap(this::onResponse, ex -> {
+            logOnFailure(LOGGER, ex);
+            onFailure(ex);
+        });
+        if (LOGGER.isDebugEnabled() == false) {
+            return listener;
+        }
         return ActionListener.wrap(r -> {
-            onResponse(r);
+            listener.onResponse(r);
             // At this point, the StopWatch should already have been stopped, so we log a consistent time.
-            LOGGER.info(
+            LOGGER.debug(
                 "Finished execution of ESQL query.\nQuery string: [{}]\nExecution time: [{}]ms",
                 esqlQuery,
                 stopWatch.stop().getMillis()
             );
         }, ex -> {
             // In case of failure, stop the time manually before sending out the response.
-            long timeMillis = stopWatch.stop().getMillis();
-            LOGGER.info("Failed execution of ESQL query.\nQuery string: [{}]\nExecution time: [{}]ms", esqlQuery, timeMillis);
-            logOnFailure(LOGGER, ex);
-            onFailure(ex);
+            if (LOGGER.isDebugEnabled()) {
```

Comment:
This line may be redundant since we only reach here when `LOGGER.isDebugEnabled() == true`, no?

Reply:
:+1: